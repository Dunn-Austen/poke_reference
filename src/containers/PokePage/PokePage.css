.poke-page {
  align-items: center;
  background-color: silver;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  height: 120vh;
  width: 100%;
}

.images-container {
  border: 1px solid green;
}

.pokepage-types {
  border: 1px solid black;
}

.formatted-type {
  text-align: center;
  text-transform: uppercase;
}

.weak-against{
  border: 1px solid purple;
}

.opponent-type {
  text-align: center;
  text-transform: uppercase;
}

.suggested-pokemon {
  border: 1px solid gold;
}

.sprite {
  height: 100px;
  width: auto;
}
