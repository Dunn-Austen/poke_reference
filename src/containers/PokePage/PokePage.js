import React, { Component } from 'react';
import './PokePage.css';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';


export const PokePage = props => {
  let inputValue;
  const { pokeData, pokeTypes } = props;

  return (
    <div className='poke-page'>
      <section className='main-left'>
        <h1>Types Container</h1>
      </section>
      <section className='main-right'>
        <section className='images-container'>
          <img
            className='sprite'
            src={pokeData.sprites.front_default}
            alt={`Front image of ${pokeData.name}`}
          />
          <img
            className='sprite'
            src={pokeData.sprites.back_shiny}
            alt={`Back image of a shiny ${pokeData.name}`}
          />
          <img
            className='sprite'
            src={pokeData.sprites.front_shiny}
            alt={`Front image of a shiny ${pokeData.name}`}
          />
          <img
            className='sprite'
            src={pokeData.sprites.back_default}
            alt={`Back image of ${pokeData.name}`}
          />
        </section>
        <section className='tactics-container'>
          <div className='weak-against'>
            

          </div>
          <div className='suggested-pokemon'>


          </div>
        </section>
      </section>
    </div>
  )
}

export const mapStateToProps = state => ({
  pokeData: state.pokeData,
  pokeTypes: state.pokeTypes
})

export default connect(mapStateToProps, null)(PokePage);

PokePage.propTypes = {
  pokeData: PropTypes.object,
  pokeTypes: PropTypes.array
}
