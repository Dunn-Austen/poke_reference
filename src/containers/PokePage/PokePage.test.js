import React from 'react';
import { shallow } from 'enzyme';
import { PokePage, mapStateToProps } from './PokePage';

describe('PokePage', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<PokePage
      pokeData={{
        abilities: [],
        height: 10,
        name: 'gengar',
        types: [],
        sprites: {
          back_default: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/1.png",
          back_shiny: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/1.png",
          front_default: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png",
          front_shiny: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/1.png"
        }
      }}
      pokeTypes={[{damage_relations: {double_damage_from: []}, pokemon: [{pokemon:{name: "abra", url: "https://pokeapi.co/api/v2/pokemon/63/"}}]}]}
      opponentTypes={[{damage_relations: {double_damage_from: []}, pokemon: [{pokemon:{name: "abra", url: "https://pokeapi.co/api/v2/pokemon/63/"}}]}]}
    />);
  });

  it('should match the NavBar Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapsStateToProps', () => {
    it('should return only the pertinent information from the redux store', () => {
      const mockState = {
        pokeNames: [{pokemon: 'raichu', url: 'url'}],
        errorMessage: '',
        isLoading: false,
        pokeTypes: [],
        pokeData: {},
        opponentTypes: [],
      };
      const expected = {
        pokeTypes: [],
        pokeData: {},
        opponentTypes: [],
      };
      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });

})
