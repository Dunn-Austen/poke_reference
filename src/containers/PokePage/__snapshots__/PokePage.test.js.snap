// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokePage should match the NavBar Snapshot 1`] = `
<div
  className="poke-page"
>
  <div
    className="inner-page"
  >
    <section
      className="pokepage-top"
    >
      <Link
        to="/"
      >
        <button
          className="home"
        >
          Return Home
        </button>
      </Link>
      <section
        className="images-container"
      >
        <img
          alt="Front side of gengar"
          className="sprite"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
        />
        <img
          alt="Back side of a shiny gengar"
          className="sprite"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/1.png"
        />
        <img
          alt="Front side of a shiny gengar"
          className="sprite"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/1.png"
        />
        <img
          alt="Back side of gengar"
          className="sprite"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/1.png"
        />
      </section>
    </section>
    <section
      className="pokepage-types"
    >
      <h1
        className="pokepage-h1"
      >
        gengar
        <span>
          's
        </span>
         Types
      </h1>
      <div
        className="organizational-container"
      />
    </section>
    <section
      className="weak-against"
    >
      <h1
        className="pokepage-h1"
      >
        Weak against
      </h1>
      <div
        className="weak-container"
      >
        <p
          className="opponent-type"
        />
      </div>
    </section>
    <div
      className="opponents-container"
    >
      <h1
        className="pokepage-h1"
      >
        Suggested Opponents
      </h1>
      <section
        className="suggested-pokemon"
      >
        <p
          className="opponent-name"
          key="abra"
        >
          abra
        </p>
      </section>
    </div>
  </div>
</div>
`;
